frontend/src/components/CheckboxGroup/CheckboxGroupValidator.tsx

```diff
@@ -0,0 +1,221 @@
+      ) {
+        setShowModal(true);
+        setModalMessage(
+          `Notice that ${query.field
+            .split('_')
+            .join(' ')} is used in a filter step, deselecting it here may break this query.`,
+        );
+      }
+      if (
+        query &&
+        query.operational_column === checkboxValue.substr(0, checkboxValue.lastIndexOf('_')) &&
+        !checkboxState &&
+        parsedOptions.id === 'aggregate'
+      ) {
+        setShowModal(true);
+        setModalMessage(
+          `Notice that ${query.operational_column
+            .split('_')
+            .join(' ')} is used in an aggregate step, deselecting it here may break this query.`,
+        );
+      }
+      if (parsedOptions.id === 'join_on') {
+        const keys = Object.keys(query.join_on);
+        for (const iterator of keys) {
+          if (iterator === checkboxValue && !checkboxState) {
+            setShowModal(true);
+            setModalMessage(
+              `Notice that ${iterator
+                .split('_')
+                .join(' ')} is used in a join step, deselecting it here may break this query.`,
+            );
+            break;
+          }
+        }
+      }
+      if (
+        query &&
+        query.operational_column === checkboxValue &&
+        !checkboxState &&
+        parsedOptions.id === 'scalar'
+      ) {
+        setShowModal(true);
+        setModalMessage(
+          `Notice that ${query.operational_column
+            .split('_')
+            .join(
+              ' ',
+            )} is used in a scalar transform step, deselecting it here may break this query.`,
+        );
+      }
+      if (query && query.operational_columns && !checkboxState && parsedOptions.id === 'multi') {
+        for (const iterator of query.operational_columns) {
+          if (iterator === checkboxValue && !checkboxState) {
+            setShowModal(true);
+            setModalMessage(
+              `Notice that ${iterator
+                .split('_')
+                .join(
+                  ' ',
+                )} is used in a multi transform step, deselecting it here may break this query.`,
+            );
+            break;
+          }
+        }
+      }
+      if (query && query.over && !checkboxState && parsedOptions.id === 'window') {
+        for (const iterator of query.columns) {
+          if (iterator === checkboxValue && !checkboxState) {
+            setShowModal(true);
+            setModalMessage(
+              `Notice that ${iterator
+                .split('_')
+                .join(' ')} is used in a window step, deselecting it here may break this query.`,
+            );
+            break;
+          }
+        }
+      }
+
+      return result;
+    }, []);
+  };
+
+  const parseOptions = (options: string) => {
+    const parsedOptions = JSON.parse(options);
+    if (parsedOptions.hasOwnProperty('filters')) {
+      const { filters }: Filters = options ? parsedOptions : { filters: [] };
```

Comment:
I find this `options` check unnecessary ... since parsedOptions is created from `options` and you're checkin to make sure that `parsedOptions` has a filter property, then this particular check is redundant. Apply elsewhere.

Reply:
I think this change is likely to get rid of all these strange default values you're adding... the whole thing looks like it was copied from somewhere.

Reply:
Updated